UnitTestFrameworkPkg/Library/UnitTestLib/UnitTestLib.c

```diff
@@ -131,6 +131,54 @@ CompareFingerprints (
   return (CompareMem (FingerprintA, FingerprintB, UNIT_TEST_FINGERPRINT_SIZE) == 0);
 }
 
+STATIC
+EFI_STATUS
+FreeUnitTestTestEntry (
+  IN UNIT_TEST_LIST_ENTRY  *TestEntry
+  )
+{
+  // TODO: Finish this function.
+  if (TestEntry) {
+    if (TestEntry->UT.Description) {
+      FreePool (TestEntry->UT.Description);
+    }
+    if (TestEntry->UT.Name) {
+      FreePool (TestEntry->UT.Name);
+    }
+    FreePool (TestEntry);
+  }
+  return EFI_SUCCESS;
+}
+
+STATIC
+EFI_STATUS
+FreeUnitTestSuiteEntry (
+  IN UNIT_TEST_SUITE_LIST_ENTRY  *SuiteEntry
+  )
+{
+  UNIT_TEST_LIST_ENTRY        *TestCase;
+  UNIT_TEST_LIST_ENTRY        *NextTestCase;
+
+  // TODO: Finish this function.
+  if (SuiteEntry) {
+    TestCase = (UNIT_TEST_LIST_ENTRY *)GetFirstNode (&(SuiteEntry->UTS.TestCaseList));
+    while ((LIST_ENTRY *)TestCase != &(SuiteEntry->UTS.TestCaseList)) {

```

Comment:
The double-casts and the manual comparison look very weird, should this not use (BASE_)CR and functions like IsNull()?